backend/server/game_api.js

```diff
@@ -0,0 +1,637 @@
+        [req.user.id, 'invited', 'accepted', 'user1_ready', 'user2_ready']);
+
+        otherGames.rows.forEach(game => {
+            if (game.state === 'invited') {
+                gameInvitations.push({
+                    game_id: game.id,
+                    opponent: game.opponent,
+                    board_dimension: game.dimension
+                });
+            }
+        });
+
+        otherGames.rows.forEach(game => {
+            if (game.state !== 'invited') {
+                activeGames.push({
+                    game_id: game.id,
+                    opponent: game.opponent,
+                    board_dimension: game.dimension,
+                    hits: 0,
+                    bombed: 0
+                });
+            }
+        });
+
+        const finalResults = {
+            user_score: {
+                wins: userStatus.rows[0].wins,
+                loses: userStatus.rows[0].loses
+            },
+            invitations: gameInvitations,
+            active_games: activeGames
+        }
+
+        return res.status(200).json(finalResults);
+    } catch (e) {
+        res.status(500).json({msg: 'Server error'});
+    }
+});
+
+
+// Update game- accept state
+gameRouter.put('/games/:game_id', async (req, res) => {
+    try {
+        const game = await pool.query('select * from games where id = $1', [req.params.game_id]);
+        const gameDetails = game.rows[0];
+
+        if (req.user.id !== gameDetails.user2) {
+            return res.status(401).json({msg: 'You are not the correct opponent player'});
+        }
+
+        if (gameDetails.state !== 'invited') {
+            return res.status(401).json({msg: 'Cannot accept an active game'});
+        }
+
+        await pool.query('update games set state = $2 where id = $1;', [req.params.game_id, 'accepted']);
+        return res.status(200).json({msg: 'game accepted by opponent'});
+
+    } catch(e) {
+        res.status(500).json({msg: 'Server error'});
+    };
+});
+
+
+// delete a game
+gameRouter.delete('/games/:game_id', async (req, res) => {
+    try {
+        const game = await pool.query('select * from games where id = $1', [req.params.game_id]);
+        const gameDetails = game.rows[0];
+
+        if (req.user.id !== gameDetails.user2) {
+            return res.status(401).json({msg: 'You are not the correct opponent player'});
+        }
+
+        if (gameDetails.state !== 'invited') {
+            return res.status(401).json({msg: 'Cannot delete an active game'});
+        }
+
+        await pool.query('delete from games where id = $1;', [req.params.game_id]);
+        return res.status(200).json({msg: 'game deleted by opponent'});
+
+    } catch(e) {
+        res.status(500).json({msg: 'Server error'});
+    };
+});
+
+
+// Placing ships
+function checkShipPlacement(start_row, end_row, start_col, end_col, shipInDb) {
```

Comment:
move it to another file. it might be the same file as `shipsAmount` and `totalShipsSizes `.